client/components/SendZoopDialog.tsx

```diff
@@ -0,0 +1,155 @@
+import React, {
+    useState,
+    useEffect 
+} from "react";
+import {useNavigate} from "react-router-dom";
+import Dialog from "@mui/material/Dialog";
+import DialogTitle from "@mui/material/DialogTitle";
+import DialogContent from "@mui/material/DialogContent";
+import DialogContentText from "@mui/material/DialogContentText"
+import DialogActions from "@mui/material/DialogActions";
+import Button from "@mui/material/Button";
+import TextField from "@mui/material/TextField";
+import Link from "@mui/material/Link";
+import Tooltip from "@mui/material/Tooltip";
+import Fab from "@mui/material/Fab";
+import SendIcon from "@mui/icons-material/Send";
+import { 
+    usePostZoopMutation, 
+    useGetAllUsersQuery,
+    useGetMeQuery
+}  from "../features/api";
+import { useSelector } from "react-redux";
+import { RootState} from "../app/store";
+import type { User } from '../../src/types/custom'
+
+const SendZoopDialog = () => {
```

Comment:
This component is not a Dialog. it is a button that contains a dialog. Recommend renaming it to SendZoopButton (especially since I also recommend creating a separate SendZoopDialog component later in the review comments, and you may also want to make a separate ZoopSentDialog component as well)